syspagen/syspagen.c

```diff
@@ -0,0 +1,784 @@
+			printf("\t\t%s\n", CAST_SYSPTR(char *, prog->argv));
+		} while ((ptr = prog->next) != sysgen_common.syspage->progs);
+	}
+	else {
+		printf("\t\tnot defined\n");
+	}
+}
+
+
+static void sysgen_help(const char *prog)
+{
+	printf("Usage: %s [options] to create binary syspage file or add syspage to image\n", prog);
+	printf("\t-s <pimg> <offs> <sz>   - syspage properties\n");
+	printf("\t    pimg - beginning physical address of the target image\n");
+	printf("\t    offs - syspage's offset in the target image\n");
+	printf("\t    sz   - max syspage's size\n");
+	printf("\t-p <path>               - path to preinit script\n");
+	printf("\t-u <path>               - path to user script\n");
+	printf("\t-i <path>               - path to image \n");
+	printf("\t-h                      - print help message\n");
+}
+
+
+int main(int argc, char *argv[])
+{
+	int opt, err;
+	char *endptr;
+	const char *preinitScript, *userScript, *binName, *imgName;
+
+	preinitScript = userScript = binName = imgName = NULL;
+
+	if (argc <= 1) {
+		sysgen_help(argv[0]);
+		return EXIT_FAILURE;
+	}
+
+	while ((opt = getopt(argc, argv, "s:p:u:i:h")) != -1) {
+		switch (opt) {
+			case 's':
+				if ((optind + 1) < argc) {
+					sysgen_common.pkernel = strtoul(argv[optind - 1], &endptr, 0);
+					if ((sysgen_common.pkernel == 0 && strlen(endptr) != 0) || (sysgen_common.pkernel != 0 && strlen(endptr) != 0)) {
+						fprintf(stderr, "Wrong physical image address %s\n", argv[optind - 1]);
+						return EXIT_FAILURE;
+					}
+
+					sysgen_common.offs = strtoul(argv[optind], &endptr, 0);
+					if ((sysgen_common.offs == 0 && strlen(endptr) != 0) || (sysgen_common.offs != 0 && strlen(endptr) != 0)) {
+						fprintf(stderr, "Wrong syspage offset %s\n", argv[optind]);
+						return EXIT_FAILURE;
+					}
+
+					sysgen_common.maxsz = strtoul(argv[optind + 1], &endptr, 0);
+					if ((sysgen_common.maxsz == 0 && strlen(endptr) != 0) || (sysgen_common.maxsz != 0 && strlen(endptr) != 0)) {
+						fprintf(stderr, "Wrong syspage size %s\n", argv[optind + 1]);
+						return EXIT_FAILURE;
+					}
+				}
+				break;
+
+			case 'p':
+				preinitScript = optarg;
+				break;
+
+			case 'u':
+				userScript = optarg;
+				break;
+
+			case 'i':
+				imgName = optarg;
+				break;
+
+			case 'h':
+				sysgen_help(argv[0]);
+				return EXIT_SUCCESS;
+
+			default:
+				return EXIT_FAILURE;
+		}
+	}
+
+	sysgen_common.buff = calloc(sysgen_common.maxsz, sizeof(uint8_t));
+	if (sysgen_common.buff == NULL)
+		return EXIT_FAILURE;
+
+	sysgen_common.syspage = (syspage_t *)sysgen_common.buff;
+	sysgen_common.syspage->size = ALIGN_ADDR(sizeof(syspage_t), sizeof(long long));
+	sysgen_common.syspage->pkernel = sysgen_common.pkernel;
+	LIST_INIT(&sysgen_common.aliases);
+
+	/* Parse preinit script with map and console commands */
+	err = sysgen_parseScript(preinitScript);
+	if (err < 0) {
+		fprintf(stderr, "Cannot parse preinit script: %s\n", preinitScript);
+		sysgen_cleanup();
+		return EXIT_FAILURE;
+	}
```

Comment:
```suggestion
	if ((sysgen_parseScript(preinitScript) < 0)
		fprintf(stderr, "Cannot parse preinit script: %s\n", preinitScript);
		sysgen_cleanup();
		return EXIT_FAILURE;
	}
```
You are not using `err` variable.
If `preinitScript  == NULL` ( missing -p option ) program prints a strange message:
```
Cannot open file (null)
Cannot parse preinit script: (null)
```

Reply:
Thanks. I have removed `err` variable and added additional checking of input parameters to not allow to printing strange messages.